File: 프로그래머스/알고리즘 고득점 Kit/완전탐색/모음사전.js
function dfs(current, length, words) {
	const vowels = [..."AEIOU"];
	if (current.length === length) {
		words.push(current);
		return;
	}
	vowels.forEach((vowel) => {
		dfs(current + vowel, length, words);
	});
}

function solution(word) {
	const words = [];
	for (let length = 1; length <= 5; length++) dfs("", length, words);
	return words.sort().indexOf(word) + 1;
}

File: 프로그래머스/알고리즘 고득점 Kit/완전탐색/모음사전.py
from itertools import product

def solution(word):
    vowels = ['A', 'E', 'I', 'O', 'U', '']
    words = set()    
    for a, b, c, d, e  in product(vowels, repeat=5):
        words.add(''.join([a, b, c, d, e]))
    words = sorted(list(words))

    return words.index(word)